block-producer/src/main.rs

```diff
@@ -0,0 +1,25 @@
+mod cli;
+mod config;
+mod server;
+use anyhow::Result;
+use clap::Parser;
+use cli::{Cli, Command};
+use config::BlockProducerConfig;
+use miden_node_utils::Config;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    miden_node_utils::logging::setup_logging()?;
+
+    let cli = Cli::parse();
+    let config: BlockProducerConfig =
+        BlockProducerConfig::load_config(cli.config.as_deref()).extract()?;
+
+    match cli.command {
+        Command::Serve { .. } => {
```

Comment:
Why does `Serve` have `port` and `host` options if we ignore them?

Reply:
Created an issue to fix this: https://github.com/0xPolygonMiden/miden-node/issues/75